### PR TITLE
[21224] Update release support table 

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description
This PR updates the release support table with the internal decisions regarding the EOL of the current LTS and supported Fast DDS branches.

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- **N/A** The code follows the style guidelines of this project. 
- **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- **N/A** Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- **N/A** Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- **N/A** Changes are API compatible. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 
    
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,15 +68,6 @@ For more information check the FIWARE Catalogue entry for [Robotics](https://git
 Write to evaluation.support@eprosima.com or mention @EProsima on Twitter.
 We are curious to get to know your use case!
 
-## Supported platforms
-
-More information about the official support can be found [here](https://github.com/eProsima/Fast-DDS/blob/master/PLATFORM_SUPPORT.md#platform-support)
```
Comment (EduPonz) [inline README.md:73]: I don't think we should remove the link to the supported platforms, just the jenkins CI badges

File: RELEASE_SUPPORT.md [hunk 2]
```diff
@@ -11,29 +11,31 @@ By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
...
+|--------------|------------|----------------------------|--------|
+|***v2.14.x (LTS)***|***March 2024***|***March 2025 [^1]***|***March 2025 [^1]***|
+|***v2.13.x***|***December 2023***|***July 2024***|***July 2024***|
+|v2.12.x|September 2023|March 2024|March 2024|
+|v2.11.x|July 2023|January 2024|January 2024|
+|***v2.10.x (LTS)***|***March 2023***|***May 2025 [^1]***|***May 2025 [^2]***|
+|v2.9.x|December 2022|July 2023|July 2023|
+|v2.8.x|September 2022|March 2023|March 2023|
+|v2.7.x|July 2022|January 2023|January 2023|
+|***v2.6.x (LTS)***|***March 2022***|***July 2024 ***|***May 2025 [^2]***|
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:28]:
```suggestion
|***v2.6.x (LTS)***|***March 2022***|***July 2024***|***May 2025 [^2]***|
```

File: RELEASE_SUPPORT.md [hunk 3]
```diff
@@ -11,29 +11,31 @@ By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
...
-|v2.4.x|September 2021|March 2022|
-|v2.3.x|March 2021|November 2022|
-|v2.2.x|January 2021|February 2022|
-|v2.1.x|November 2020|May 2023|
-|v2.0.x|June 2020|February 2022|
-|v1.10.x|April 2020|February 2022|
-|v1.9.x|August 2019|February 2022|
-|v1.8.x|May 2019|February 2022|
-|v1.7.x|December 2018|February 2022|
+|Version branch|Release Date|End of Standard Support Date|EOL Date|
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:18]: I think it's time to split this table into two, one for  supported and eol versions

File: RELEASE_SUPPORT.md [hunk 4]
```diff
@@ -11,29 +11,31 @@ By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
...
+|v2.1.x|November 2020|May 2023|May 2023|
+|v2.0.x|June 2020|February 2022|February 2022|
+|v1.10.x|April 2020|February 2022|February 2022|
+|v1.9.x|August 2019|February 2022|February 2022|
+|v1.8.x|May 2019|February 2022|February 2022|
+|v1.7.x|December 2018|February 2022|February 2022|
 
-[^*]: Support may be extended.
+[^1]: Support may be extended.
+[^2]: Critical and security fixes support restricted until EOL Date.
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:41]: I'm not sure I understand this

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -53,5 +58,6 @@
 - [ ] The PR has a milestone assigned.
 - [ ] The title and description correctly express the PR's purpose.
 - [ ] Check contributor checklist is correct.
+- [ ] If critical bug fix, backport to the 2.6.x branch has been opened.
```
Comment (EduPonz) [inline .github/pull_request_template.md:61]:
```suggestion
- [ ] If this is a critical bug fix, backports to the critical-only supported branches has been requested.
```

File: RELEASE_SUPPORT.md [hunk 6]
```diff
@@ -11,20 +11,29 @@ By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
 However, the life of certain branches can be extended depending on costumer and user needs.
 Within the support period of any minor version, there may be several patch releases that either add new functionalities in an ABI compatible way or fix possible issues.
 Before the minor release end of life (EOL) date, a final patch release including the latest fixes will be made available if there are any changes not included in any previous patch release of the same minor version.
+Long Term Supported (LTS) versions may have an extended supported period for only critical issues and security fixes, where no other ABI compatible neither bug fixes will be backported.
+This period applies since the end of standard support date to the end of life (EOL) date.
 
-The table below outlines the *eProsima Fast DDS* minor releases and their support cycles:
+The table below outlines the *eProsima Fast DDS* currently supported versions and their support cycles:
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:17]: I'd make a title instead

File: RELEASE_SUPPORT.md [hunk 7]
```diff
@@ -11,20 +11,29 @@ By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
...
+|Version branch|Release Date|End of Standard Support Date|EOL Date|
+|--------------|------------|----------------------------|--------|
+|v2.14.x (LTS)|March 2024|March 2025 [^*]|March 2025 [^*]|
+|v2.13.x|December 2023|July 2024|July 2024|
+|v2.10.x (LTS)|March 2023|May 2025 [^*]|May 2025|
+|v2.6.x (LTS)|March 2022|July 2024|May 2025|
+
+[^*]: Support may be extended.
+
+Moreover, the following table records the lifecycle of previously supported Fast DDS versions.
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:28]: Same here

File: RELEASE_SUPPORT.md [hunk 8]
```diff
@@ -11,20 +11,29 @@ By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
 However, the life of certain branches can be extended depending on costumer and user needs.
 Within the support period of any minor version, there may be several patch releases that either add new functionalities in an ABI compatible way or fix possible issues.
 Before the minor release end of life (EOL) date, a final patch release including the latest fixes will be made available if there are any changes not included in any previous patch release of the same minor version.
+Long Term Supported (LTS) versions may have an extended supported period for only critical issues and security fixes, where no other ABI compatible neither bug fixes will be backported.
+This period applies since the end of standard support date to the end of life (EOL) date.
 
-The table below outlines the *eProsima Fast DDS* minor releases and their support cycles:
+The table below outlines the *eProsima Fast DDS* currently supported versions and their support cycles:
+
+|Version branch|Release Date|End of Standard Support Date|EOL Date|
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:19]: I propose two more columns to be added to these tables (on the left):

1. Version (like `2.14`, `2.13`, and so on)
2. Latest release (like `v2.14.2`, linked with the corresponding release notes)

File: RELEASE_SUPPORT.md [hunk 9]
```diff
@@ -11,20 +11,29 @@ By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
...
+|--------------|------------|----------------------------|--------|
+|v2.14.x (LTS)|March 2024|March 2025 [^*]|March 2025 [^*]|
+|v2.13.x|December 2023|July 2024|July 2024|
+|v2.10.x (LTS)|March 2023|May 2025 [^*]|May 2025|
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:23]: EOL here should be October 2024

File: RELEASE_SUPPORT.md [hunk 10]
```diff
@@ -11,20 +11,29 @@ By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
...
+|v2.6.x (LTS)|March 2022|July 2024|May 2025|
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:24]: The support for this may also be extended

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: RELEASE_SUPPORT.md (hunk 8 above)
Comment (JesusPoderoso) [reply RELEASE_SUPPORT.md]: I've updated the release checklist to update it when new version released

File: .github/pull_request_template.md [hunk 11]
```diff
@@ -53,5 +58,6 @@
 - [ ] The PR has a milestone assigned.
 - [ ] The title and description correctly express the PR's purpose.
 - [ ] Check contributor checklist is correct.
+- [ ] If this is a critical bug fix, backports to the critical-only supported branches has been requested.
```
Comment (EduPonz) [inline .github/pull_request_template.md:61]:
```suggestion
- [ ] If this is a critical bug fix, backports to the critical-only supported branches have been requested.
```

File: RELEASE_SUPPORT.md [hunk 12]
```diff
@@ -11,20 +11,29 @@ By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
...
 Before the minor release end of life (EOL) date, a final patch release including the latest fixes will be made available if there are any changes not included in any previous patch release of the same minor version.
+Long Term Supported (LTS) versions may have an extended supported period for only critical issues and security fixes, where no other ABI compatible neither bug fixes will be backported.
+This period applies since the end of standard support date to the end of life (EOL) date.
 
-The table below outlines the *eProsima Fast DDS* minor releases and their support cycles:
+## *eProsima Fast DDS* currently supported versions and their support cycles
+
+|Version|Version branch|Latest Release|Release Date|End of Standard Support Date|EOL Date|
+|-------|--------------|--------------|------------|----------------------------|--------|
+|2.14|v2.14.x (LTS)|[v2.14.2](https://fast-dds.docs.eprosima.com/en/v2.14.2/notes/notes.html)|March 2024|March 2025 [^*]|March 2025 [^*]|
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:21]: Apply to all

```suggestion
|2.14|[2.14.x](https://github.com/eProsima/Fast-DDS/tree/2.14.x) (LTS)|[v2.14.2](https://fast-dds.docs.eprosima.com/en/v2.14.2/notes/notes.html)|March 2024|March 2025 [^*]|March 2025 [^*]|
```

File: RELEASE_SUPPORT.md [hunk 13]
```diff
@@ -11,20 +11,29 @@ By default, *eProsima Fast DDS* minor releases have a lifecycle of **6 months**.
...
+|2.13|v2.13.x|[v2.13.5](https://fast-dds.docs.eprosima.com/en/v2.13.5/notes/notes.html)|December 2023|July 2024|July 2024|
+|2.10|v2.10.x (LTS)|[v2.10.4](https://fast-dds.docs.eprosima.com/en/v2.10.4/notes/notes.html)|March 2023|October 2024 [^*]|May 2025|
+|2.6|v2.6.x (LTS)|[v2.6.8](https://fast-dds.docs.eprosima.com/en/v2.6.8/notes/notes.html)|March 2022|July 2024|May 2025[^*]|
+
+[^*]: Support may be extended.
+
+## *eProsima Fast DDS* previously supported versions.
```
Comment (EduPonz) [inline RELEASE_SUPPORT.md:28]: This table should have the same format as the previous

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]

**Pull request merged**